user/test/com/google/gwt/emultest/java10/util/stream/CollectorsTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.emultest.java10.util.stream;
+
+import static com.google.gwt.emultest.java8.util.stream.CollectorsTest.applyItems;
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static java.util.stream.Collectors.toUnmodifiableMap;
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+import com.google.gwt.emultest.java.util.EmulTestBase;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Tests for java.util.stream.Collectors Java 10 API emulation.
+ */
+public class CollectorsTest extends EmulTestBase {
+  private static <T> boolean unmodifiableCollection(Collection<T> c, T existingSample, T newSample) {
+    try {
+      c.remove(existingSample);
+      return false;
+    } catch (Exception ignore) {
+      // expected
+    }
+    try {
+      c.add(newSample);
+      return false;
+    } catch (Exception ignore) {
+      // expected
+    }
+    Iterator<T> itr = c.iterator();
+    itr.next();
+    try {
+      itr.remove();
+      return false;
+    } catch (Exception e) {
+      // expected
+    }
+    return true;
+  }
+
+  public void testToUnmodifiableList() {
+    applyItems(List.of("a", "b"), toUnmodifiableList(), "a", "b", (a, b) -> {
```

Comment:
All these tests are fine but a bit difficult to review/read. `applyItems` isn't a great name as it also executes tests using the provided equals function. I had to lookup what `applyItems` actually does. Also using `(a, b)` was a bit distracting as `a` and `b` are used as values as well. Using `expected` and `actual` would probably make more sense. Then we also don't need to test immutability of `expected`.

Reply:
Happy to have a proposed name change for `applyItems()`, but this has gone through 4ish patches now, and I think the Javadoc is fairly clear - this method applies the collector to the items specified in a way that validates that the Collector instance functions according to its contract. A more accurate name would probably be too unwieldy, and still require that the test author/reader check for themselves what it does (`applyCollectorToItemsWithAndWithoutSplittingAndValidateResultsMatchExpected`?), or we could make it only marginally more clear - but this would still require the reader to actually check the code to understand the implications (`applyCollectorToItemsAndValidate`, `applyCollectorAndValidate`, `validateCollectorBehavior`?)

I'll rename the lambda args.

Reply:
> applyCollectorToItemsWithAndWithoutSplittingAndValidateResultsMatchExpected
Hehe..

I thought about unwrapping it, e.g.
```
var list1 = collectItemsWithSplitting(collector, items...)
var list2 = collectItemsWithoutSplitting(collector, items...)
assertAllImmutable(list1, list2);
assertAllEqual(expected, list1, list2)
```
or
```
var result = collectItems(collector, items...)
assertAllImmutable(result.withSplitting, result.withoutSplitting);
assertAllEqual(expected, result.withSplitting, result.withoutSplitting)
```